spec/facility_locations_spec.rb

```diff
@@ -0,0 +1,66 @@
+
+missouri = {"name":"OAKVILLE","address1":"3164 TELEGRAPH ROAD","city":"ST LOUIS","state":"MO","zipcode":"63125","county":"St. Louis County","phone":"(314) 887-1050","fax":"(314) 887-1051","size":"3","email":"OAKVILLEAGENTOFFICE@DOR.MO.GOV"}
+# I did clip some of tha data off from oakville so that data wasnt taking up so much space. hits all the same points.
+
+describe FacilityLocation do
+
+
+  it 'parses address to string' do
+    expect(locations.parsed_address(oregon)).to eq('2242 Santiam Hwy SE Albany OR 97321')
+    expect(locations.parsed_address(new_york)).to eq('200 E. MAIN STREET STE. 101 ROCHESTER NY 14604')
+    expect(locations.parsed_address(missouri)).to eq('3164 TELEGRAPH ROAD ST LOUIS MO 63125')
+  end
+
+
+  it 'formats data to hash' do
+    oregon_formatted = [{name: 'Albany DMV Office', address: '2242 Santiam Hwy SE Albany OR 97321', phone: '541-967-2014'}]
+    new_york_formatted = [{:address=>'200 E. MAIN STREET STE. 101 ROCHESTER NY 14604', :name=>'ROCHESTER DOWNTOWN', :phone=>'5857531604'}]
+    missouri_formatted = [{:name=>'OAKVILLE', :address=>'3164 TELEGRAPH ROAD ST LOUIS MO 63125', :phone=>'(314) 887-1050'}]
+
+    expect(locations.formatted_location(oregon)).to eq(oregon_formatted)
+    expect(locations.formatted_location(new_york)).to eq(new_york_formatted)
+    expect(locations.formatted_location(missouri)).to eq(missouri_formatted)
+  end
+
+
+  it 'creates data form oregan api' do
+    or_dmv_office_locations = DmvDataService.new.or_dmv_office_locations
+    or_dmv = locations.create_facility(or_dmv_office_locations)
+    expect(or_dmv).to be_a(Array)
+    expect(or_dmv.first).to be_a(Facility)
+    expect(or_dmv.first.name).to eq('Albany DMV Office')
+    expect(or_dmv.first.address).to eq('2242 Santiam Hwy SE Albany OR 97321')
+    expect(or_dmv.first.phone).to eq('541-967-2014')
+  end
+
+  #did check for other locations in the files to show that it isnt pulling from already set vars
+  it 'creates data form new york api' do
+    ny_dmv_office_locations = DmvDataService.new.ny_dmv_office_locations
+    ny_dmv = locations.create_facility(ny_dmv_office_locations)
+    expect(ny_dmv).to be_a(Array)
+    expect(ny_dmv[1]).to be_a(Facility)
+    expect(ny_dmv[1].name).to eq('ROCHESTER DOWNTOWN')
+    expect(ny_dmv[1].address).to eq('200 E. MAIN STREET STE. 101 ROCHESTER NY 14604')
+    expect(ny_dmv[1].phone).to eq('5857531604')
+  end
```

Comment:
We'll talk more about class methods in a later class, but with the way you have this `create_facility` set up, it might be a good time to use a class method. I choose to use a class method when I want functionality in a class, but don't care much about state. Take a look into class methods if you'd like!